### ISSUE TITLE
Dependency error when integrating Victoire in a Symfony project

### ISSUE THREAD

**Issue by alexislefebvre** [Victoire/victoire#691]
## Description

I created a brand new Symfony 3.0 project: 

    $ composer create-project symfony/framework-standard-edition symfonyVictoire "3.0.*" -vvv --profile --no-progress
    cd symfonyVictoire

Then I tried to add Victoire by following the steps in the [setup guide](https://github.com/Victoire/victoire/blob/master/Bundle/CoreBundle/Resources/doc/SETUP.md):

    $ composer require victoire/victoire

But this returns an error:

    […]
    Your requirements could not be resolved to an installable set of packages.
      Problem 1
        - victoire/victoire 2.2.1 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.2.0 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - Installation request for victoire/victoire ^2.2 -> satisfiable by victoire/victoire[2.2.0, 2.2.1].
    
    Installation failed, deleting ./composer.json.

## Resolution Ideas
It looks like a previous issue: #390.

**Comment by alexislefebvre** [comment]
Specifying a minimal version gives similar results:

    $ composer require victoire/victoire "^2.0" -vvv --profile --no-progress
    […]
    ./composer.json has been updated
    Loading composer repositories with package information
    Updating dependencies (including require-dev)
    Your requirements could not be resolved to an installable set of packages.

      Problem 1
        - victoire/victoire 2.2.1 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.2.0 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.1.9 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.1.8 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - […]
        - victoire/victoire 2.0.2 requires doctrine/doctrine-fixtures-bundle 2.2.x-dev -> satisfiable by doctrine/doctrine-fixtures-bundle[2.2.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.0.1 requires doctrine/doctrine-fixtures-bundle 2.2.x-dev -> satisfiable by doctrine/doctrine-fixtures-bundle[2.2.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.0.0 requires doctrine/doctrine-fixtures-bundle 2.2.x-dev -> satisfiable by doctrine/doctrine-fixtures-bundle[2.2.x-dev] but these conflict with your requirements or minimum-stability.
        - Installation request for victoire/victoire ^2.0 -> satisfiable by victoire/victoire[2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.1.0, 2.1.1, 2.1.10, 2.1.11, 2.1.2, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.1.7, 2.1.8, 2.1.9, 2.2.0, 2.2.1].


    Installation failed, reverting ./composer.json to its original content.

**Comment by alexislefebvre** [comment]
The reason is that Victoire is not compatible with Symfony 3 at this moment.

**Comment by alexislefebvre** [comment]
I have a similar error when trying to adding Victoire in a Symfony 2.8 project:

    > composer require victoire/victoire
    Using version ^2.2 for victoire/victoire
    ./composer.json has been updated
    Loading composer repositories with package information
    Updating dependencies (including require-dev)
    Your requirements could not be resolved to an installable set of packages.
    
      Problem 1
        - victoire/victoire 2.2.1 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - victoire/victoire 2.2.0 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
        - Installation request for victoire/victoire ^2.2 -> satisfiable by victoire/victoire[2.2.0, 2.2.1].
    
    
    Installation failed, reverting ./composer.json to its original content.

The error comes from FOSJsRoutingBundle which doesn't have a stable release for the 2.0 branch yet: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/issues/262

**Comment by alexislefebvre** [comment]
Here is the content of the `composer.json` file of one of my projects:

```json
{
    "name": "symfony/framework-standard-edition",
    "license": "MIT",
    "type": "project",
    "description": "The \"Symfony Standard Edition\" distribution",
    "autoload": {
        "psr-4": { "": "src/"},
        "classmap": [ "app/AppKernel.php", "app/AppCache.php" ]
    },
    "require": {
        "php": ">=5.3.3",
        "symfony/symfony": "2.8.*",
        "doctrine/orm": "^2.4.8",
        "doctrine/doctrine-bundle": "~1.4",
        "symfony/assetic-bundle": "~2.3",
        "symfony/swiftmailer-bundle": "~2.3",
        "symfony/monolog-bundle": "~2.4",
        "sensio/distribution-bundle": "~5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "~2.0",
        "twig/extensions": "~1.0",
        "symfony/icu": "v1.1.0",
        "sonata-project/admin-bundle" : "~3.0",
        "sonata-project/doctrine-orm-admin-bundle": "~3.0",
        "victoire/victoire": "~2.2"
    },
    "require-dev": {
        "sensio/generator-bundle": "~3.0",
        "symfony/phpunit-bridge": "~2.7",
        "phpunit/phpunit": "~5.1",
        "doctrine/doctrine-fixtures-bundle": "~2.2",
        "liip/functional-test-bundle": "~1.4"
    },
    "scripts": {
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ]
    },
    "config": {
        "bin-dir": "bin"
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-web-dir": "web",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        },
        "branch-alias": {
            "dev-master": "2.8-dev"
        }
    }
}
```

I have an error when I try to `install` or `update` my dependencies with Composer, even if I remove the `composer.lock` file and `vendor/` directory.

But if I copy and paste this file in an empty directory, then everything can be install without any error. I don't understand what's going on.

**Comment by alexislefebvre** [comment]
I moved this to https://github.com/Troopers/AlertifyBundle/issues/66

**Comment by alexislefebvre** [comment]
I updated my test protocol:

```bash
$ composer create-project symfony/framework-standard-edition symfonyVictoire "~2.8" --no-interaction
Installing symfony/framework-standard-edition (v2.8.16)
  - Installing symfony/framework-standard-edition (v2.8.16) Loading from cache
Created project in symfonyVictoire
[…]
$ cd symfonyVictoire
$ composer require "victoire/victoire:~2.2"
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - victoire/victoire 2.2.9 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.8 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.7 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.6 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.5 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.4 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.3 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.2 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.14 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.13 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.12 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.11 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.10 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.1 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - victoire/victoire 2.2.0 requires friendsofsymfony/jsrouting-bundle ~2.0 -> satisfiable by friendsofsymfony/jsrouting-bundle[2.0.0-alpha1, 2.0.x-dev] but these conflict with your requirements or minimum-stability.
    - Installation request for victoire/victoire ~2.2 -> satisfiable by victoire/victoire[2.2.0, 2.2.1, 2.2.10, 2.2.11, 2.2.12, 2.2.13, 2.2.14, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9].


Installation failed, reverting ./composer.json to its original content.
```

**Comment by talbotseb** [comment]
@alexislefebvre We've been reported this dependency problem to cope with Sylius : synmfony/monolog-bundle should be on ^3.0 (as in Sylius) whereas it is ~2.4 in Victoire

**Comment by alexislefebvre** [comment]
I tried to allow `dev` version of packages but I still can't install Victoire in a fresh Symfony 2.8 project:

```bash
composer create-project symfony/framework-standard-edition symfonyVictoire "2.8.*" --no-interaction ; cd symfonyVictoire
# Allow "dev" packages
composer config "minimum-stability" "dev"
# Update minimal version of PHP 
composer config "platform.php" "5.5.9"
composer require victoire/victoire:"~2.2"
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for doctrine/orm (locked at v2.4.8, required as ^2.4.8) -> satisfiable by doctrine/orm[v2.4.8].
    - victoire/victoire 2.2.0 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.1 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.10 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.11 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.12 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.13 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.14 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.15 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.2 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.3 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.4 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.5 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.6 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.7 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.8 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.9 requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - victoire/victoire 2.2.x-dev requires doctrine/orm v2.5.2 -> satisfiable by doctrine/orm[v2.5.2].
    - Conclusion: don't install doctrine/orm v2.5.2
    - Installation request for victoire/victoire ~2.2 -> satisfiable by victoire/victoire[2.2.0, 2.2.1, 2.2.10, 2.2.11, 2.2.12, 2.2.13, 2.2.14, 2.2.15, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.2.x-dev].
```

**Comment by alexislefebvre** [comment]
The problem has been fixed, thanks to #795. It works if I `require` the dependencies listed in README.md: https://github.com/Victoire/victoire/blob/571a1541de5c861aafc4d3ad762d671ee7c3fd9b/doc/setup.md#composer

```bash
composer create-project symfony/framework-standard-edition symfonyVictoire "2.8.*" --no-interaction
cd symfonyVictoire
# "composer create-project" defines an old version of PHP, overwrite it
composer config "platform.php" "5.5.9"
composer require victoire/victoire friendsofsymfony/user-bundle:~2.0@dev doctrine/orm --update-with-dependencies
```
